### DISCUSSION TITLE
Need help adding a Solis Inverter over RS485 using Waveshare RS485 over eth (b)

### DISCUSSION THREAD

**Discussion by dimi-be** [wills106/homeassistant-solax-modbus#568] [General]
Hello,

I'm trying to setup this integration for my Solis RHI-3K-48ES-5G using a Waveshare RS485 over eth (b). I can't get it to work I think I've tried every possible settings configuration I could imagine, but no success. I even tried switching the RS485 A en B wire.

I keep getting the following error: "unrecognized Solis inverter type - serial number : unknown". I found that this is related to a communication error, but I think my wiring is correct (it's only 2 wires and I tried switching them).

What might be special about my setup is that I also have Eastron power meter that is connected over RS485. The Eastron has address 1 and I've set the address of the inverter to 2. I think the Eastron also needs to have address 1 because I cannot find a way to change this address in the settings of the inverter.

I've added the wires of the Waveshare in parallel over the wires coming from the inverter and going to the Eastron. I think this is ok, because the communication between the inverter and the power meter keeps working.

On the Waveshare the leds light up like this: pwr = red, link = green and act = green and blue flashes (short blue flashes and long green, when I switched the A and B wire it was long blue flashes and short green). I can access the Waveshare gui on the network.

I have this feeling that I'm missing something in the settings because of a lack of knowledge on my end or that it is the Eastron meter that is complicating things. I tried directly connecting to the inverter, without the power meter, but I'm getting the same result.

I hope someone can help. I've added screenshots of my settings.

Regards,
Dimitri

![wavehare-settings1](https://github.com/wills106/homeassistant-solax-modbus/assets/11840783/645e20f4-7cd6-489c-a1fc-11399ac23b6f)
![wavehare-settings2](https://github.com/wills106/homeassistant-solax-modbus/assets/11840783/b52477f8-6376-4517-b515-684bb3c34de4)
![ha-settings-1](https://github.com/wills106/homeassistant-solax-modbus/assets/11840783/cfdbe7fe-9a0a-4dc2-b1ca-2cd5ae14174d)
![ha-settings-2](https://github.com/wills106/homeassistant-solax-modbus/assets/11840783/3e418ea5-4bc2-45c3-8811-e4c1c564bf83)

**Comment by alienatedsec** [comment]
@dimi-be 

Please select `Modbus TCP` and make sure you have the right IP address as you are trying `192.168.10.72` instead of `192.168.1.72`

![image](https://github.com/wills106/homeassistant-solax-modbus/assets/73167064/c927b08e-84f6-4ffd-a097-0f14e295bfc4)

**Reply by dimi-be** [reply]
Thank you for your reply. I first had it on separate network that has no internet connection. It switched it over to my main network for easier testing. I think this was typo after my many times of trying and setting up the screens for screenshots from scratch. I tried setting it  up like you say but I still get the same error.

![ha-settings-3](https://github.com/wills106/homeassistant-solax-modbus/assets/11840783/61fd14fa-54d7-47ab-8cec-087c2d2a2610)
![error-1](https://github.com/wills106/homeassistant-solax-modbus/assets/11840783/09346dee-aba2-4801-9592-d8adc39325e5)

**Reply by alienatedsec** [reply]
@dimi-be 

I am not sure what you amended on your inverter, but the only settings that can be changed are related to the COM port. You need to check your inverter address and baud rate settings on the inverter and use the same in your integration.

**Reply by dimi-be** [reply]
I think I'm connecting to the wrong port. I'm connected to the port labelled as RS485 meter, but I think I need to connect to the one labelled com that currently has my wifi stick attached. Does someone have experience with this before I try and hunt down the correct connector?

**Reply by alienatedsec** [reply]
@dimi-be check this https://github.com/alienatedsec/solis-ha-modbus-cloud#devices-and-cost

Also, read the whole repo as it will tell more about your connectivity options.

**Reply by dimi-be** [reply]
Thank you. So it isn't as easy as I was hoping. Are the client and server Waveshares needed? Can't I just hook into the same bus as where the wifi stick is or will this create collision issues? I'm guess it is otherwise you wouldn't have done it that way :)

**Reply by alienatedsec** [reply]
> Can't I just hook into the same bus as where the wifi stick is or will this create collision issues? I'm guess it is otherwise you wouldn't have done it that way :)

If you want to run both - a Solis DataLogger like `S3-WIFI-ST` and connect with the Home Assistant - you need additional hardware e.g., as described in my repo - there are also other ways of connecting using "2 masters 1 slave" type of hardware, but it is hit and miss with revisions and it needs the additional hardware anyway.

If you have any Modbus TCP-compatible datalogger e.g., `S2-WL-ST` or `DLS-L` you can still use this integration without any additional hardware - or `DLS-W` to use with other (solarman) integration.

With the `S3-WIFI-ST` - there are other GitHub repos e.g., https://github.com/hn/ginlong-solis which modified its firmware, but you are likely to disable SolisCloud for good by reflashing it.

**Reply by dimi-be** [reply]
I have the S3-WIFI-ST and I would like to keep using it and not modify it. I'm planning on renting or selling the house it's currently in in the next couple of years, so all changes to it need to be reversible.

So it looks like my only option is to get a second wavelink, box and the connector and build what you've build.

This inverter came cheap because I bought it in a good group buying deal. If I ever get a new installation I would do some research before so that I get one that can do everything I need it to do.

Thank you for helping me and for writing a great guide on how to do what I need. I really appreciate it! I will try to not forget to update here once I get everything working.

And to anyone who has a similar problem as me, check that you're connecting through the COM port and not the meter port 🙈

**Comment by kshales** [comment]
this is my settings.
![image](https://github.com/wills106/homeassistant-solax-modbus/assets/77297717/ba1cac5a-8c94-45e1-9458-62a715a699d7)

**Reply by kshales** [reply]
ive found that if i update the integration i cant get any option to work now.
this is the integration i use. dont think you can download it now.
is your inverter modbus address really  = 2 ? have you changed it, they usually default 1
 
![image](https://github.com/wills106/homeassistant-solax-modbus/assets/77297717/c3234838-4c62-4911-b125-d8674a262c49)

**Reply by dimi-be** [reply]
@kshales thank you for taking the time to respond, but my issue was something different. I was connecting to the wrong port on the invert. I connected to the "meter" port instead of the "com" port. I was confused because the port said RS-485.

@alienatedsec, @kshales It took me a while after findin the above out to get to this fase, but (today) I finally had the time to put all the extra components together to tap into the right port and to connect to the Solis inverter over modbus while keeping the data logger. It's all working and I can see the inverter and all the sensors in HA now!

Thank you @alienatedsec for writing up all the information! After wiring everything up and setting the waveshare up as in your guide it worked from the first try!

**Reply by alienatedsec** [reply]
@dimi-be the community is huge, and there are even more projects once you have it up and running :)

**Reply by dimi-be** [reply]
@alienatedsec this is indeed the beginning of the journey. I have many plans for this in the near future (mainly peak shaving and startegic loading with dyanmic tariffs) :)
